test/SecTester.Scan.Tests/ScanSettingsTests.cs

```diff
@@ -0,0 +1,237 @@
+  }
+
+  public void Dispose()
+  {
+    _targetOptions.ClearSubstitute();
+    _scanSettingsOptions.ClearSubstitute();
+  }
+
+  [Fact]
+  public void PublicConstructor_CreatesInstanceWithDefaultOptions()
+  {
+    // act
+    var result = new ScanSettings(_targetOptions, new List<TestType> { TestType.Csrf });
+
+    // assert
+    result.Should().BeEquivalentTo(new
+    {
+      RepeaterId = null as string,
+      Name = "GET example.com",
+      PoolSize = 10,
+      TargetTimeout = TimeSpan.FromSeconds(5),
+      SlowEpTimeout = TimeSpan.FromSeconds(1000),
+      Smart = true,
+      SkipStaticParams = true,
+      Tests = new List<TestType> { TestType.Csrf },
+      AttackParamLocations = new List<AttackParamLocation>
+      {
+        AttackParamLocation.Body, AttackParamLocation.Query, AttackParamLocation.Fragment
+      },
+      Target = new { Url = TargetUrl }
+    });
+  }
+
+  [Fact]
+  public void InternalConstructor_CreatesInstance()
+  {
+    // act
+    var result = new ScanSettings(_scanSettingsOptions);
+
+    // assert
+    result.Should().BeEquivalentTo(new
+    {
+      RepeaterId,
+      Name = ScanName,
+      PoolSize = 1,
+      TargetTimeout = TimeSpan.FromSeconds(5),
+      SlowEpTimeout = TimeSpan.FromSeconds(200),
+      Smart = false,
+      SkipStaticParams = false,
+      Tests = new List<TestType> { TestType.Csrf },
+      AttackParamLocations = new List<AttackParamLocation> { AttackParamLocation.Query },
+      Target = new { Url = TargetUrl }
+    });
+  }
+
+  [Fact]
+  public void InternalConstructor_NameIsNull_CreatesInstanceWithDefaultName()
+  {
+    // arrange
+    _scanSettingsOptions.Name.ReturnsForAnyArgs(_ => null!);
+
+    // act
+    var result = new ScanSettings(_scanSettingsOptions);
+
+    // assert
+    result.Should().BeEquivalentTo(new { Name = "GET example.com" });
+  }
+
+  [Fact]
+  public void InternalConstructor_NameIsNull_CreatesInstanceTruncatedHost()
+  {
+    // arrange
+    _scanSettingsOptions.Name.ReturnsForAnyArgs(_ => null);
+    _scanSettingsOptions.Target.Url.ReturnsForAnyArgs($"https://{new string('a', 200)}.example.com/api/v1/info");
+
+    // act
+    var result = new ScanSettings(_scanSettingsOptions);
+
+    // assert
+    result.Should().BeEquivalentTo(new { Name = $"GET {new string('a', 195)}…" });
+  }
+
+  [Fact]
+  public void InternalConstructor_CreatesInstanceWithUniqueAttackParamLocations()
+  {
+    // arrange
+    _scanSettingsOptions.AttackParamLocations.ReturnsForAnyArgs(
+      new List<AttackParamLocation> { AttackParamLocation.Header, AttackParamLocation.Header });
+
+    // act
+    var result = new ScanSettings(_scanSettingsOptions);
+
+    // assert
+    result.Should().BeEquivalentTo(new
+    {
+      AttackParamLocations = new List<AttackParamLocation> { AttackParamLocation.Header }
+    });
+  }
+
+  [Fact]
+  public void InternalConstructor_CreatesInstanceWithUniqueTests()
```

Comment:
ditto (you want to say that this case is not applicable for other constructors?)